src/sync.rs

```diff
@@ -225,6 +227,7 @@ where
     stopping: bool,
     state: ActorState,
     factory: Arc<dyn Fn() -> A>,
+    address: Addr<A>,
```

Comment:
I think `crate::address::AddressSenderProducer` would be more suited to be used here instead of `Addr`. The type shows that it doesn't intend to send messages to the Actor itself, but that it can produce address for it. It is used in the [ContextParts](https://github.com/actix/actix/blob/master/src/contextimpl.rs#L42) if you wanna look at some code examples.

Reply:
Yes, the general idea is the same. The `address` method doesn't need to change it's signature at all, just the type internally used by the `SyncContext`. I think it'll make code clearer and similar to how other internal parts of actix works.

Since the SyncContext itself won't be sending message to it's `Actor` it doesn't need to hold an `Addr`. But it does need (as this PR implements) a way to produce new address to it's Actor so whomever access access to the context can also send messages to the Actor, and that's what an `AddressSenderProducer` is for.